pkg/metrics/statsd/statsd_test.go

```diff
@@ -83,3 +85,240 @@ func TestEmitFloat(t *testing.T) {
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			s := &statsd{}
+			part1, part2, err := s.parseSocketEnv(tt.teststring)
+
+			if tt.wantError {
+				if err == nil {
+					t.Error(fmt.Errorf("expected error but didn't get one."))
+				}
+			} else {
+				if err != nil {
+					t.Error(fmt.Errorf("unexpected error received: %q ", err.Error()))
+				}
+				if part1 != tt.part1 {
+					t.Error(fmt.Errorf("part1 does not match: Wanted: %q but got %q ", tt.part1, part1))
+				}
+				if part2 != tt.part2 {
+					t.Error(fmt.Errorf("part2 does not match: Wanted: %q but got %q ", tt.part2, part2))
+				}
+			}
+		})
+	}
+}
+
+func TestValidateSocketDefinition(t *testing.T) {
+	for _, tt := range []struct {
+		name         string
+		network      string
+		address      string
+		expectToPass bool
+		wantError    bool
+	}{
+		{
+			name:         "Valid UDP case",
+			network:      "udp",
+			address:      "127.0.0.1:9000",
+			expectToPass: true,
+			wantError:    false,
+		},
+		{
+			name:         "Valid Unix Domain Socket case",
+			network:      "unix",
+			address:      "/var/something/or/another",
+			expectToPass: true,
+			wantError:    false,
+		},
+		{
+			name:         "Invalid protocoll",
+			network:      "tcp",
+			address:      "127.0.0.1:12",
+			expectToPass: false,
+			wantError:    true,
+		},
+		{
+			name:         "Valid protocol, random invalid address - is good here",
+			network:      "udp",
+			address:      "somerandomtext149203#$%",
+			expectToPass: true,
+			wantError:    false,
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			s := &statsd{}
+
+			ok, err := s.validateSocketDefinition(tt.network, tt.address)
+
+			if tt.wantError && err == nil {
+				t.Error(fmt.Errorf("Test %s, expected error but didn't get one.", tt.name))
+			}
+
+			if !tt.wantError && err != nil {
+				t.Error(fmt.Errorf("Test %s, unexpected validation error %q.", tt.name, err.Error()))
+			}
+
+			if ok != tt.expectToPass {
+				t.Error(fmt.Errorf("Test %q,unexpected validation result: Expected: %t, Got %t ", tt.name, tt.expectToPass, ok))
+			}
+		})
+	}
+}
+
+func TestGetConnectionDetails(t *testing.T) {
+	for _, tt := range []struct {
+		name string
+
+		isLocalDev           bool
+		mdmsocketstring      string
+		network              string
+		address              string
+		wantErrorToStartWith string
+	}{
+		{
+			name:                 "Old behaviour / production mode",
+			isLocalDev:           false,
+			mdmsocketstring:      "",
+			network:              "unix",
+			address:              "/var/etw/mdm_statsd.socket",
+			wantErrorToStartWith: "",
+		},
+		{
+			name:                 "Old Behaviour / localdev mode",
+			isLocalDev:           true,
+			mdmsocketstring:      "",
+			network:              "unix",
+			address:              "mdm_statsd.socket",
+			wantErrorToStartWith: "",
+		},
+		{
+			name:                 "Valid UDP env variable",
+			isLocalDev:           false,
+			mdmsocketstring:      "udp:127.0.0.1:9000",
```

Comment:
This is not standard URL format is it ? To me it would be `udp://127.0.0.1:9000`

Reply:
@facchettos related comment: https://github.com/Azure/ARO-RP/pull/2074#discussion_r856261859